tests/test_apdb.py

```diff
@@ -330,10 +327,7 @@ def test_storeForcedSources(self):
         objects = _makeObjectCatalogPandas(region, 100, config)
         catalog, oids = _makeForcedSourceCatalogPandas(objects)
 
-        apdb.storeDiaObjects(objects, visit_time)
-
-        # save them
-        apdb.storeDiaForcedSources(catalog)
+        apdb.store(visit_time, objects, forced_sources=catalog)
```

Comment:
I cannot say, I'd expect AP pipeline experts to know more about that. My expectation is that when you have new DiaSource or DiaForcedSource this also results in a new version of corresponding DiaObject, so very likely you always have objects. The code in this class will actually fail if you try to save sources without their corresponding objects because pixelization (and partitiong in case of Cassandra) relies on object's ra/dec.

For profiling I do expect that you will get terrible result with sqlite, there is probably no point in even trying to profile or improve that. Or did you do it with Postgres?

Reply:
My point was not that we want to use anything in particular for profiling, just that I/O looks to be a significant algorithmic bottleneck. But if you think we can't ever use objects as-is then that's good enough for me.

Reply:
If there is a use case for storing sources separately from their objects we need to think how to pass object ra/dec information without passing whole object catalog. Should not be too hard, just let me know what kind of API works best for you 🙂